w06-ruby-on-rails/reading-intro-to-class.md

```diff
@@ -0,0 +1,395 @@
+# Introduction to Classes
+
+### Object vs. Class
+
+An object represents an abstract thing, usually with some properties (attributes) and some sort of behavior (methods). A class, in turn, can create many objects with the same set of properties (attributes) and behaviors (methods).
+
+### Define and Instantiate a Class
+
+Class definitions require three basic components: the reserved word `class`, a name of the class, and the reserved word `end`. By convention, the class name begins with a capital letter.
+
+```ruby
+class Car
+end
+```
+
+To create instances of our class, we will create a variable and assign it the return value of the `Car` class's `Car.new` method.
+
+```ruby
+car = Car.new
+```
+
+### Instance methods and instance variables
+
+We are able to create instances of a `Car`, but our class is *very* simple and our instances currently don't give us much. We'd like to add properties and behaviors (attributes and methods) for our car instances. We'll start by looking at **instance methods** and **instance variables**. An instance method represents a function that is accessible on every instance of a class. To create an instance method, we just create a regular method inside our class definition. Here's the syntax:
+
+```ruby
+class Car
+  def drive
+    puts "You're going places!"
+  end
+end
+```
+
+Now every car we create will have a "drive" behavior.
+
+```ruby
+car.drive
+```
+
+Let's give each instance of `Car` a color using instance variables.
+
+An instance variable -- which begins with the `@` symbol -- has the capability of storing data for each instance of a class.
+
+JavaScript lets us access variables inside objects with syntax like `obj.var` or `obj["var"]`, similar to javascript. In Ruby, we'll create getter and setter methods by hand instead of accessing instance variables directly. We can then access the getters and setters with dot notation.  Let's look at an example.
+
+```ruby
+class Car
+  def color=(color)
+    @color = color
+  end
+
+  def color
+    @color
+  end
+end
+
+car_one = Car.new
+car_one.color = "green"
+car_one.color
+
+car_two = Car.new
+car_two.color = "black"
+car_two.color
```

Comment:
I think I would move the `car_one.color` and `car_two.color` down to the bottom to reinforce the fact that each instance retains its own color.